### DISCUSSION TITLE
Is the term "custom directory" confusing?

### DISCUSSION THREAD

**Discussion by gilgongo** [jamulussoftware/jamulus#2100] [UI Related]
I find I think of what we currently call a "custom directory" as being a "private directory". This is because there's nothing customised about them (they don't have any different features) and like private _servers_, you have to know what their address is before you can access the (also private) servers on them. 

Of course, this issue hardly affects very many people, but I thought I'd ask. It may be that my "mental model" about this isn't representative.

**Comment by pljones** [comment]
"Private" is simply wrong, so it's not an option here.  There may be nothing private about it.  "Custom" just means "not built in".

Like "custom content" in games is stuff you don't get built into the game, and is often supplied by other players.

**Reply by gilgongo** [reply]
But you can't find a custom directory without its address in the same way as you can't find a private server without it's address. Hence my question about confusion in the term "custom". What is "custom" about it?

**Reply by pljones** [reply]
"Private" is a subset of "custom".  Not all custom directories are private.  All private directories are custom.

**Reply by pljones** [reply]
Use Case 1:

User A runs a music blog and they decide to host a number of servers.  They want to attract users to their blog so they set up a custom directory and register the servers with that.  The directory is _not private_ -- anyone anywhere on the internet could find it with a quick search engine look up -- but it is custom.

Use Case 2:

A Company wants to market their product by hosting a number of servers where the product will be used.  But they only want users of the product to use it.  So to get the address, you need to register with their site and then your IP gets into the server firewall whitelist.  Unless you're a paid up, registered user, you don't have access to the directory.  That's a private directory and hosting private servers.  Of course, it's still custom.

**Reply by gilgongo** [reply]
Indeed, and both of those use cases might apply equally well if you swapped out "directory" for "server" ("The _server_ is not private -- anyone anywhere...") - so why don't we call private servers "custom servers"?

But my point here is not about the details of the use cases, but the potential for confusion about the term "custom" when people _do not_ know the use cases. Might Jamulus users encounter the term "custom directory" and think they can have a directory that acts differently to a normal one, has different features, etc. in the same way as a customised car? In which case, they might go looking in the docs or online expecting information about that and finding it hard to understand what they encounter? This might be more or less acute in other languages, for all I know.

So my wondering is whether the word "private" may be more descriptive of the semantic "domain" that custom servers are in, as it were.

**Reply by pljones** [reply]
Servers are servers.  They're what the client sees in the server list for a Directory or enters in the connect dialog Server Address.  They're public if registered in a public directory (or if the unregistered address is publicised).  That public directory could be a custom directory.  They're private if registered in a private directory (which would be a custom directory) or not registered and the address is kept private.  It's not the same.

**Reply by ghost** [reply]
configurable directory, configurable directories.

**Reply by pljones** [reply]
See also https://github.com/jamulussoftware/jamulus/pull/2079#issuecomment-951370812

**Reply by gilgongo** [reply]
"Configurable" may be a bit less confusing, but I would say that seeing as we already have the concept of something being "private" in Jamulus as one where you need to configure the client to see it by adding an address in the connect dialogue, then if you have to add an address in the settings for directories, then it would seem logical to apply that word to directories too. Using the word "custom" implies they may work differently, so I wonder if people start off on the wrong foot when they read the docs. Documentation confusion often stems from mistaken assumptions about the meaning of terms. Where do we tell people how to customise their directories?

But like I say, this probably only affects a very few people. Would be interesting to see what others think though.

**Reply by pljones** [reply]
"Configuration", to me, relates to settings.  So if it's only used in relationship to what's set up there, it makes sense.

If "any input field" used by the program becomes "configuration", then I think it loses its meaning.  So care required.

Given that "custom" has been in use as the term for a directory in which to register that is not provided by the Jamulus software ever since the feature was introduced, I think it's likely _less_ confusing to leave it alone.

**Reply by ghost** [reply]
Maybe the 'Custom Directories' item could be the 'Directories' item populated (jamulus.ini) with the built-in directory servers. This way the directory servers are not compiled into Jamulus and 'Custom' is no longer relevant.

**Reply by gilgongo** [reply]
> I think it's likely less confusing to leave it alone.

Only if we assume the same people will be using Jamulus forever, but yes in the short to medium term :-)

**Reply by pljones** [reply]
> 
> 
> Maybe the 'Custom Directories' item could be the 'Directories' item populated (jamulus.ini) with the built-in directory servers. This way the directory servers are not compiled into Jamulus and 'Custom' is no longer relevant.

I've considered this.  We've very few places where we provide a "Restore _this specific setting_ to default values".  This one would be critical.  I've no strong objections to it, though.  My view is that there really shouldn't be a concept of "centrally owned servers" which the current built-in set continues to encourage (hence the old "Central" terminology).  It should be more about the Jamulus user community providing their own resources and working together to co-ordinate.  That's a long way off realisation, though.
